### ISSUE TITLE
ARG network: DiGraph or MultiDiGraph

### ISSUE THREAD

**Issue by hyanwong** [tskit-dev/what-is-an-arg-paper#70]
In https://github.com/tskit-dev/what-is-an-arg-paper/blob/a0234149698295d058fb6ef0ea8331039ea67722/argutils/viz.py#L31

we convert the tree sequence into a graph. This should at least be a Digraph, and we have [argued before] that it should probably also be a MultiDigraph (my feeling is that the true representation of the TS as a graph should be a multidigraph potentially with multiple intervals per edge too, and I think @petrelharp agrees.

Either way, for viz we need to turn it into a multidigraph, I think. Perhaps we should have an option in `convert_nx` to return a multidigraph (or I can do the conversion from digraph -> multidigraph in the `draw` function, I guess). Which do you prefer @jeromekelleher ?

**Comment by jeromekelleher** [comment]
I don't think it matters very much @hyanwong - what's the difference between having a multigraph of edges [(u, v, I1),  ..., (u, v, Ik)] vs a graph of edges [(u, v, [I1, ..., Ik])]? From a graph theoretic perspective I prefer the latter as it's less redundant and easier to explain.

I don't think there's much information in having multiple edges between a pair of nodes for an ARG - the only time you'll have more than one is when there's trapped material. A simplified tree sequence is a different story, but maybe we should worry about that separately.

I'd prefer to focus on visualising the quantitative aspects of spans and coalescences along edges before we spend more time on the graph-vs-multigraph thing.

**Comment by hyanwong** [comment]
> I'm just trying to re-do the plot for #32 using the new code, and I need to plot multiple lines between 2 nodes for that.

I might have misunderstood here though: you are right that it's probably only the simplified version that requires multiple edges between the same 2 nodes.

**Comment by jeromekelleher** [comment]
> I think each edge should represent a path of transmission: this is what I reckon is understood by an edge.

I agree that's what a **tree sequence** edge is, but let's just try to be more graph theoretic about it for a bit and see where we get to.

> I'm just trying to re-do the plot for #32 using the new code, and I need to plot multiple lines between 2 nodes for that.

Can we try it without this first? I don't want to start optimising the plotting code before we look at edge weights.

**Comment by hyanwong** [comment]
Sure. Just to check, in your new code:

https://github.com/tskit-dev/what-is-an-arg-paper/blob/5c6255dc6f0396e6504fde486e52d05ede702c4c/argutils/ancestry.py#L196

How do we represent a diamond? Do we have a node attached to 2 parent pass-though nodes, which are themselves attached to a coalescent node?

It's mainly the diamonds that I want to represent sensibly in the viz.

**Comment by jeromekelleher** [comment]
It's not an issue any more because we have two parents now, not one.

**Comment by hyanwong** [comment]
Right: I was confused because I thought parents might count as pass-through nodes, and therefore be dropped as per https://github.com/tskit-dev/what-is-an-arg-paper/blob/5c6255dc6f0396e6504fde486e52d05ede702c4c/argutils/ancestry.py#L308.

**Comment by hyanwong** [comment]
I think we need an issue about whether to use a MultiDiGraph for simplified ARGs / tree sequences. I could change the title of this one, or close and open another.

**Comment by hyanwong** [comment]
> > I think each edge should represent a path of transmission: this is what I reckon is understood by an edge.
> 
> I agree that's what a **tree sequence** edge is, but let's just try to be more graph theoretic about it for a bit and see where we get to.

I'm not at all sure that is what a tree sequence edge is, TBH. We often have multiple edges that have gone through the same transmission path, don't we?

**Comment by jeromekelleher** [comment]
I don't think it matters here for the purposes of this paper - we're not making software, so it's just semantics.

**Comment by hyanwong** [comment]
I think it might be relevant to the viz in #82. Probably worth discussing over a cup of tea?

**Comment by hyanwong** [comment]
Returning to this issue, and copying in @awohns, who had expressed an interest. I'm starting to think that whether or not the graph is a Digraph or a MultiDigraph, we want to *visualize* each (tree sequence) edge as a separate line. This makes things much easier when we remove pass-through nodes, and also it means that we can illustrate the breaking-up of regions into separate edges, without having to label the edges themselves, which gets very confusing. It also helps when weighting the blackness of the line by the span of the edge.

**Comment by jeromekelleher** [comment]
I don't think we need an issue open for this - if we want to viz the fully resolved ts as a multigraph then we can.
